pkg/controller/pxc/controller.go

```diff
@@ -1109,3 +1099,40 @@ func createOrUpdateConfigmap(cl client.Client, configMap *corev1.ConfigMap) erro
 
 	return nil
 }
+
+func updateServiceIfNeeded(cl client.Client, old, new *corev1.Service) error {
```

Comment:
I think this func should not update service. It just should return true of false depend on is services are equal or not.

Reply:
068a20ad